### DISCUSSION TITLE
Retrive getStepId in Closure Job

### DISCUSSION THREAD

**Discussion by wit3** [ksassnowski/venture#66] [Q&A]
Hi, thanks for this awesome package.

I've a question.
How can retrive getStepId() in a closure added in workflow?

For a simply Job it's possibile using getStepId from trait `WorkflowStep`

But when i use this how can access? 

```php
$this->define('My Workflow')
    ->addJob(function (TruthGenerator $truths) {
        //access here to stepid or workflow_id
      	Log::info($truths->getHotTake());
    }, id: 'log-truth');
```

**Comment by wit3** [comment]
Maybe implementing parameters in `ClosureWorkflowStep` class?

https://github.com/ksassnowski/venture/blob/40a5bb1367cd61d8ac9da203a7b8b67134043e4b/src/ClosureWorkflowStep.php#L32

propose :

```php
app()->call($this->callback->getClosure(), ['workflow' => $this->workflow(), 'step' => $this->step()]);
```

with this it's possibile to use

```php
$this->define('My Workflow')
    ->addJob(function (TruthGenerator $truths, WorkflowJob $step,Workflow $workflow) {
        Log::info('workflow id: '. $workflow->id);
        Log::info('step name: '. $step->name);
      	Log::info($truths->getHotTake());
    }, id: 'log-truth');
```

Sorry for my basic implementation, tell me if need PR or suggestion

**Reply by stevebauman** [reply]
I think if we passed in the whole `ClosureWorkflowStep` job, that would provide the most utility:

```php
app()->call($this->callback->getClosure(), ['job' => $this]);

```

```php
$this->define('...')
    ->addJob(function (TruthGenerator $truths, ClosureWorkflowStep $job) {
        // ...
    });
```

**Reply by wit3** [reply]
make a PR with this implementation #67
